cdap-ui/app/cdap/services/cdapavscwrapper/DecimalLogicalType.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import AbstractLogicalType, { IJsonResponse } from 'services/cdapavscwrapper/AbstractLogicalType';
+import { LogicalTypes } from 'services/cdapavscwrapper/LogicalTypes';
+import cdapavsc from 'cdap-avsc';
+
+const UNDERLYING_TYPE = 'bytes';
+interface IDecimalJsonResponse extends IJsonResponse {
+  precision?: number;
+  scale?: number;
+}
+const JSON_FORMAT: IDecimalJsonResponse = {
+  type: UNDERLYING_TYPE,
+  logicalType: LogicalTypes.DECIMAL,
+};
+
+export default class DecimalLogicalType extends AbstractLogicalType {
+  constructor(attrs, opts) {
+    if (attrs.precision) {
+      JSON_FORMAT.precision = attrs.precision;
+    }
+    if (attrs.scale) {
+      JSON_FORMAT.scale = attrs.scale;
+    }
```

Comment:
I think this will be lost when exportType is called. These logical type classes is not a singleton, a new instance will get created when the parser encounter this logical type. Perhaps change the JSON_FORMAT to be a private variable of the class?

Reply:
hmm actually private variable won't work either since it's a static method.... hmmm

Reply:
This is strange. The `exportType` is never called for the logical type mappings we have. In the `formatOutput` call when we get the type from avsc it returns an object. BUT the right JSON is returned in `toJSON` call that gets handled in the `AbstractLogicalType` class. In this case since its a class function the right JSON is returned.

This is the reason I was not seeing any issues during import/export of pipeline and schema in individual nodes.

Reply:
<img width="1130" alt="Screen Shot 2019-05-08 at 10 47 57 AM" src="https://user-images.githubusercontent.com/1452845/57396305-00d11b00-717f-11e9-9f86-aa136e8284e9.png">